pkg/resources/kafka/kafka.go

```diff
@@ -699,15 +710,19 @@ func (r *Reconciler) reconcileKafkaPod(log logr.Logger, desiredPod *corev1.Pod,
 			return errorfactory.New(errorfactory.StatusUpdateError{}, statusErr, "updating per broker config status for resource failed", "kind", desiredType)
 		}
 
-		if val, ok := r.KafkaCluster.Status.BrokersState[desiredPod.Labels["brokerId"]]; ok && val.GracefulActionState.CruiseControlState != v1beta1.GracefulUpscaleSucceeded {
-			gracefulActionState := v1beta1.GracefulActionState{ErrorMessage: "CruiseControl not yet ready", CruiseControlState: v1beta1.GracefulUpscaleSucceeded}
+		if val, hasBrokerState := r.KafkaCluster.Status.BrokersState[desiredPod.Labels["brokerId"]]; hasBrokerState {
+			ccState := val.GracefulActionState.CruiseControlState
+			incompleteDownscale := ccState == v1beta1.GracefulDownscaleRequired || ccState == v1beta1.GracefulDownscaleRunning
```

Comment:
Couldn't `incompleteDownscale` be replaced with `ccState.IsDownscale()` ?

Reply:
I think it can be used. It is not the same, but it should works. 
func (r CruiseControlState) IsDownscale() bool {
	return r == GracefulDownscaleRequired || r == GracefulDownscaleSucceeded || r == GracefulDownscaleRunning
}
First the kafafkaPodDelete() run and it will remove the broker from status with GracefulDownscaleSucceeded. So to check the  r == GracefulDownscaleSucceeded part is unnecessary. It is safe to use the isDownscale() I'm going to change the implementation.

Reply:
Checking `GracefulDownscaleSucceeded` is still needed to avoid unnecessary update.